src/goto-symex/field_sensitivity.h

```diff
@@ -34,7 +34,18 @@ class field_sensitivityt
   /// index.
   /// \param state: symbolic execution state
   /// \param lhs: non-expanded symbol
-  void field_assignments(goto_symex_statet &state, const exprt &lhs);
+  void field_assignments(goto_symex_statet &state, const ssa_exprt &lhs);
+
+  /// If `is_divisible(lhs)` then assign indivisible symbols (fields or array
+  /// indices) from the divisible symbol and erase the divisible symbol from
+  /// \p state.
+  /// \param state: symbolic execution state
+  /// \param lhs: non-expanded symbol
+  /// \param ns: symex's working namespace
+  void post_process_assignment(
```

Comment:
not a great name, I'm not sure what it should be, something like  `assign_indivisible_and_erase_divisible`?

Reply:
Does the user actually need to know what it exactly does? The user should know when to invoke it (after the assignment -> post process), but not necessary what exactly is done in there (which might actually change).